STM32MP2 Series support

youssefz24 opened the pull request:
Add basic support of the Cortex-M33 core for the STM32MP2 series.
Provide support for the STM32MP257F-EV1 evaluation board.

It depends on the following PR from the STM32-HAL:https://github.com/zephyrproject-rtos/hal_stm32/pull/282

This first step allows to run and debug the samples:
- Hello_world
- Blinky
- Button

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
dts: arm: st: stm32mp25*_m33.dtsi: add init dtsi

Add the initial device tree source include (dtsi) files for the
stm32mp25 series boards, covering non-secure configuration for zephyr on
the Cortex-M33 core.
These files provide the basic hardware description, including CPU
(Cortex-M33), memory, RCC clock controller and NVIC interrupt
controller.

Key features:
- Set flash and RAM addresses to DDR memory.
- Adjust RCC peripheral address for non-secure context.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
soc: stm32: add initial soc support for stm32mp2x

Add initial soc support for the stm32mp2x series, including
initial Kconfig entries and default configuration files.
This enables Zephyr to recognize and build for the stm32mp2x series,
taking the stm32mp257f_ev1 as a baseline.

Includes:
- Kconfig and defconfig files for SoC selection and defaults
- soc.h for hal headers
- CMakeLists.txt for build system integration
- soc.yml update to register the new SoC

System Clock is configured statically from DTS. So no initialization
hook or soc.c needed.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
boards: st: add support for stm32mp257f_ev1 board

Introduce initial board support for the STM32MP257F-EV1. This includes:
- Board metadata (board.yml, stm32mp257f_ev1.yaml)
- Base software configuration (Kconfig.stm32mp257f_ev1)
- CMake build integration (board.cmake)
- Default configuration (stm32mp257f_ev1_stm32mp257fxx_m33_defconfig)
- Minimal device tree (stm32mp257f_ev1_stm32mp257fxx_m33.dts)
- Initial documentation and image for the stm32mp257f_ev1 board.
- OpenOCD configuration files for debugging support.

This enables the STM32MP257F-EV1 board to build and debug a hello world
sample.

Note that other necessary openocd configuration files are added to the
zephyr sdk directory and can be found in the stm-openocd repository[1].
[1]:https://github.com/STMicroelectronics/device-stm-openocd/tree/main

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
drivers: interrupt_controller: add stm32mp2 exti

Add the mp2 exti2 dts to the dtsi file.
Add mp2 exti hal and ll function calls with EXTI2 instance. We use the
EXTI2 instance because it contains the GPIO interrupts in the non-secure
context. (We are trying to build the blinky sample as a first milestone)

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
drivers: gpio: add mp2 gpio clock handling

Remove gpio clock management from the GPIO driver when running on the
cortex-m33 on the mp2 and gpio clocks are managed by the cortex-A, being
the resource manager, allowed by the Resource Isolation Framework (RIF).
Also add a specific binding for the mp2 gpio to make clock property
optional.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
drivers: clock_control: add mp2 clock driver

Add the stm32mp2 clock driver to the clock_control subsystem. The driver
is a reduced version of the generic stm32 clock driver.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
dts: clock_control: add mp2 rcc binding

Introduce DeviceTree binding for the STM32MP2 RCC clock controller,
enabling support for STM32MP2-specific clock configuration in Zephyr.
Update Kconfig.stm32 to add a dependency on STM32MP2 configuration,
allowing the use of STM32 LL RCC features when targeting STM32MP2
devices.
Add header for STM32MP2 per peripheral clock definitions.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
soc: linker.ld: add linker script for stm32mp2x

Add a linker script for the stm32mp2x soc series. It includes the
standard arm cortex-m linker and adds standard zephyr relocation
sections.
Replace the rom_start section name with .isr_vectors in the linker
script. This is necessary for the zephyr firmware to be started by the
remote proc driver which expects the section containing the vector table
to be named .isr_vectors.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
boards: add gpio, button and led support

Add gpio support for STM32MP257F-EV1 board.
Add the orange (&led0) LED to the device tree and enable GPIOJ.
Add the user button USER2 to the device tree and enable GPIOG.
This step allows building and debugging the blinky and button sample
applications.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 committed:
dts: arm: st: stm32mp2_m33.dtsi: add GPIO nodes

Add GPIO A-K nodes to the device tree for STM32MP2 SoC.
Note that GPIOs are disabled by default in the STM32MP2 SoC.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

erwango commented on soc/st/stm32/stm32mp2x/CMakeLists.txt:
```diff
@@ -9,4 +9,4 @@ zephyr_sources(
 zephyr_include_directories(.)
 
-set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/arm/cortex_m/scripts/linker.ld CACHE INTERNAL "")
+set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
```
Let's remove this line from previous commit and avoid fix in this commit

erwango commented on boards/st/stm32mp257f_ev1/stm32mp257f_ev1.dts:
To ease introduction of CortexA support, see other multicore boards (stm32h747i_disco or other vendors) to see how to add core variants. Check also Zephyr HW model documentation.

youssefz24 replied:
Done

erwango commented on drivers/interrupt_controller/intc_exti_stm32.c:
```diff
@@ -290,7 +306,29 @@ void stm32_gpio_intc_select_line_trigger(stm32_gpio_irq_line_t line, uint32_t tr
+
 	switch (trg) {
+#if defined(CONFIG_SOC_SERIES_STM32MP2X)
+	case STM32_GPIO_IRQ_TRIG_NONE:
+	LL_EXTI_DisableRisingTrig_0_31(EXTI2, exti_line);
+	LL_EXTI_DisableFallingTrig_0_31(EXTI2, exti_line);
+	break;
+	case STM32_GPIO_IRQ_TRIG_RISING:
+	LL_EXTI_EnableRisingTrig_0_31(EXTI2, exti_line);
+	LL_EXTI_DisableFallingTrig_0_31(EXTI2, exti_line);
+	break;
+	case STM32_GPIO_IRQ_TRIG_FALLING:
+	LL_EXTI_EnableFallingTrig_0_31(EXTI2, exti_line);
+	LL_EXTI_DisableRisingTrig_0_31(EXTI2, exti_line);
+	break;
+	case STM32_GPIO_IRQ_TRIG_BOTH:
+	LL_EXTI_EnableRisingTrig_0_31(EXTI2, exti_line);
+	LL_EXTI_EnableFallingTrig_0_31(EXTI2, exti_line);
+	break;
```
indentation

youssefz24 replied:
Done

erwango commented on dts/arm/st/mp2/stm32mp257.dtsi:
```diff
@@ -61,6 +63,101 @@
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x44240000 DT_SIZE_K(1)>;
+				clocks = <&rcc STM32_CLOCK(GPIOA, CLK)>;
```
Because the `clocks` binding differs from other STM32, it would be required to have a dedicated MP2 RCC compatible to explain how to interpret various fields.

erwango commented on drivers/clock_control/clock_stm32_ll_mp2.c:
```diff
@@ -25,10 +25,9 @@ static inline int stm32_clock_control_on(const struct device *dev,
-	* See (for example) RM0440 7.2.17
+	/*
+	* Note: GPIO clocks are actually enabled by the cortex-A35 and are
+	* protected by the RIF. Enabling them here would only trigger an IAC event
```
To be fixed in initial commit.

erwango commented on boards/st/stm32mp257f_ev1/doc/stm32mp257f_ev1.rst:
```diff
@@ -184,12 +184,12 @@ Zephyr firmware using OpenOCD.
 
 - Copy the firmware to the board, load it and start it with remoteproc
-  (`STM32MP257F boot Cortex-M33 firmware`_)
+  (`STM32MP257F boot Cortex-M33 firmware`_). The orange LED should be blinking.
```
Please fix all this in initial commit which introduces the doc

youssefz24 replied:
Thanks for reply.
This was intended to be an update of the doc. At the initial doc commit only hello_world was supported. But now that blinky is supported I updated the doc for it. Should I introduce the doc at this point instead and avoid having a fix ?

erwango reviewed (changes requested):
Thanks @youssefz24 for this work!

Some initial comments, but looks globally fine!

arnopo commented:
You should add the update of  of `west.yml` that point to  to https://github.com/zephyrproject-rtos/hal_stm32/pull/282

See https://docs.zephyrproject.org/latest/develop/modules.html#process-for-submitting-changes-to-existing-modules for details

arnopo commented on dts/arm/st/mp2/stm32mp257.dtsi:
```diff
@@ -0,0 +1,48 @@
+/*
```
to rename stm32mp257_m33.dtsi

arnopo commented on soc/st/stm32/stm32mp2x/Kconfig:
```diff
@@ -0,0 +1,16 @@
+	select ARM
+	select ARMV8_M_DSP
+	select ARM_TRUSTZONE_M
+	select CPU_CORTEX_M33
```
Right we should find code related to the M33 in  
`soc/st/stm32/stm32mp2x/m33 subfolder`

erwango replied:
> Right we should find code related to the M33 in `soc/st/stm32/stm32mp2x/m33 subfolder`

Yeah, but only some parts. Kconfig stuff should remain there I think.

youssefz24 replied:
I made the necessary changes. Let me know if the naming should be changed ?

nordicjm commented on boards/st/stm32mp257f_ev1/Kconfig.defconfig:
```diff
@@ -0,0 +1,11 @@
+
+if BOARD_STM32MP257F_EV1
+
+config BOARD
+        default "stm32mp257f_ev1"
```
hwmv1 was dropped years ago, file to go

youssefz24 replied:
Done

nordicjm commented on boards/st/stm32mp257f_ev1/stm32mp257f_ev1.dts:
```diff
@@ -0,0 +1,21 @@
+#include <st/mp2/stm32mp257.dtsi>
+
+/ {
+    model = "STMicroelectronics STM32MP257F-EV1 board";
```
tab indents

nordicjm commented on boards/st/stm32mp257f_ev1/stm32mp257f_ev1.dts:
```diff
@@ -0,0 +1,21 @@
+/ {
+    model = "STMicroelectronics STM32MP257F-EV1 board";
+	compatible = "st,stm32mp257f-ev1", "st,stm32mp25";
+	chosen {
```
line 13 on new line

nordicjm commented on boards/st/stm32mp257f_ev1/board.cmake:
```diff
@@ -1,4 +1,7 @@
 # Copyright (C) 2025 Savoir-faire Linux, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+board_runner_args(openocd "--config=${BOARD_DIR}/support/openocd.cfg")
```
squash with previous commit

nordicjm commented on boards/st/stm32mp257f_ev1/doc/stm32mp257f_ev1.rst:
```diff
@@ -0,0 +1,223 @@
+
+Overview
+********
+
```
squash with previous commit

youssefz24 replied:
Done

nordicjm commented on drivers/clock_control/clock_stm32_ll_mp2.c:
```diff
@@ -0,0 +1,89 @@
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+
+static inline int stm32_clock_control_on(const struct device *dev,
+	clock_control_subsys_t sub_system)
```
indentation is off

nordicjm commented on drivers/clock_control/clock_stm32_ll_mp2.c:
```diff
@@ -0,0 +1,89 @@
+}
+
+static inline int stm32_clock_control_off(const struct device *dev,
+	 clock_control_subsys_t sub_system)
```
etc. fix for whole file (put it through clang-format?)

nordicjm commented on drivers/clock_control/clock_stm32_ll_mp2.c:
```diff
@@ -0,0 +1,89 @@
+	ARG_UNUSED(dev);
+
+	if (IN_RANGE(pclken->bus, STM32_PERIPH_BUS_MIN, STM32_PERIPH_BUS_MAX) == 0) {
+	/* Attempt to toggle a wrong periph clock bit */
+	return -ENOTSUP;
```
etc. no more pointed out

youssefz24 committed:
west.yml: update modules hal stm32 for stm32mp2x

Update west.yml to add the STM32MP2x SoC support in the STM32 HAL.
This is done after the merge of the PRs:
https://github.com/zephyrproject-rtos/hal_stm32/pull/282
https://github.com/zephyrproject-rtos/hal_stm32/pull/286

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 commented on dts/arm/st/mp2/stm32mp257.dtsi:
```diff
@@ -0,0 +1,48 @@
+			clocks-controller;
+			#clock-cells = <2>;
+			reg = <0x44200000 DT_SIZE_K(64)>;
+			label = "STM32_CLK_RCC";
```
Done

youssefz24 commented on soc/st/stm32/stm32mp2x/Kconfig.defconfig.stm32mp25_m33:
Done

arnopo reviewed (changes requested):
I only review until commit https://github.com/zephyrproject-rtos/zephyr/pull/90295/commits/2ff8bc20304c5a16b9cdd2d204547fed51107402

MrVyM commented on soc/st/stm32/stm32mp2x/m33/soc.h:
```diff
@@ -0,0 +1,25 @@
+
+#endif /* !_ASMLANGUAGE */
+
+#endif /* _STM32MP2SOC_H_ */
```
```suggestion
#endif /* !_STM32MP2SOC_H_ */
```

youssefz24 commented on boards/st/stm32mp257f_ev1/doc/stm32mp257f_ev1.rst:
```diff
@@ -0,0 +1,223 @@
+side.
+- Renaming the 'rom_start' section to '.isr_vectors' in the zephyr build
+system. *This is the solution used for now.* The section name is changed in the
+linker script located in :zephyr_file:`soc/st/stm32/stm32mp2x/m33/linker.ld`
```
In that case I would suggest only mentioning the "renaming" for now as it is functional. and update with the ENTRY() directive when it becomes available on the Linux side. Let me know if this works out for you.

youssefz24 commented on dts/arm/st/mp2/stm32mp257_m33.dtsi:
```diff
@@ -0,0 +1,47 @@
+/*
```
For now I didn't tackle this but I can if you deem it necessary to have a good file architecture to facilitate support for other boards of the mp2 series. If so, please give me more details on how you would like the files to be organized ?

youssefz24 commented on boards/st/stm32mp257f_ev1/doc/stm32mp257f_ev1.rst:
```diff
@@ -0,0 +1,223 @@
+
+Programming and Debugging
+*************************
+
```
Hope this is satisfying this way. If more details are deemed necessary please let me know.

youssefz24 commented on boards/st/stm32mp257f_ev1/doc/stm32mp257f_ev1.rst:
```diff
@@ -0,0 +1,223 @@
+===============
+
+- Cortex®-M33 with TrustZone®
+- Cortex®-M0+ for secure boot
```
I tried updating some of the fields. If it isn't enough I can always do a formatted copy paste from the ST page. Please do let me know. :smiley:

youssefz24 commented on soc/st/stm32/stm32mp2x/m33/soc.c:
```diff
@@ -0,0 +1,28 @@
+#include <zephyr/init.h>
+#include <soc.h>
+
+#include <cmsis_core.h>
```
Done

youssefz24 commented on soc/st/stm32/stm32mp2x/m33/soc.h:
```diff
@@ -0,0 +1,25 @@
+ * Chapter 2.4.2: Memory map and register boundary addresses
+ */
+
+#ifndef _STM32MP2SOC_H_
```
I m guessing it was about the "#endif" so I added the "!" mark.

arnopo reviewed (commented):
Please find a new set of comments, including a review of all commits.

youssefz24 commented on drivers/interrupt_controller/intc_exti_stm32.c:
```diff
@@ -82,6 +85,9 @@ static inline int stm32_exti_is_pending(stm32_gpio_irq_line_t line)
 #elif defined(CONFIG_SOC_SERIES_STM32H7X) && defined(CONFIG_CPU_CORTEX_M4)
 	return LL_C2_EXTI_IsActiveFlag_0_31(line);
+#elif defined(CONFIG_SOC_SERIES_STM32MP2X)
+	return (LL_EXTI_IsActiveRisingFlag_0_31(EXTI2, stm32_exti_linenum_to_src_cfg_line(line)) ||
```
Its because of the data types, putting directly `line` as an argument triggers assert_fails on the HAL functions.
Since the` linenum` is the same as `cfg_line` for the mp2, I pass it through this function. 
Another way would be to change in the `stm32_exti_isr` function (line 150) to directly have `line = line_num`.

arnopo replied:
I'm quite suppressed that it can not  work as stm32mp13, that uses `st,stm32g0_exti` device tree compatible for the exti. But I perhaps missed something? If is the case could you give details on the issue you are facing based on a concrete example?

youssefz24 commented on drivers/clock_control/clock_stm32_ll_mp2.c:
```diff
@@ -0,0 +1,87 @@
+	/*
+	 * Note: GPIO clocks are actually enabled by the cortex-A35 and are
+	 * protected by the RIF. Enabling them here would only trigger an IAC event
+	 */
```
I actually tried to have a gpio binding for the mp2, blocking the clocks property. and tinkering a bit with the gpio driver to remove clock initialization. but it ended up being more complicated than just having a if test in the clock driver saying to do  nothing if its gpio. that's why i went with that solution. Should I try to make it work with the specific binding instead ?

etienne-lms commented on soc/st/stm32/stm32mp2x/m33/soc.h:
```diff
@@ -0,0 +1,25 @@
+ */
+
+#ifndef _STM32MP2_M33_SOC_H_
+#define _STM32MP2_M33_SOC_H_
```
Nitpicking: 
```patch
-_STM32MP2_M33_SOC_H_
+_STM32MP2X_M33_SOC_H_
```
to better relate to the file path.

nordicjm commented on drivers/clock_control/Kconfig.stm32:
```diff
@@ -2,6 +2,7 @@
 
 # Copyright (c) 2017 Linaro
 # Copyright (c) 2017 RnDity Sp. z o.o.
+# Copyright (C) 2025 Savoir-faire Linux, Inc.
```
copyright added to file with no changes?

arnopo reviewed (commented):
Only 2 remaining points that need to be clarify:
- exti driver: it is strange to me that we have to implement specific code.
- board/stm32mp25x_dk.cfg needs to be provided or at least need to explain how to install it for debugging the platform

erwango commented on dts/bindings/gpio/st,stm32mp2-gpio.yaml:
```diff
@@ -0,0 +1,47 @@
+      in that case would only trigger an IAC event.
+      If the Cortex-M core is designated as the resource manager, then the
+      clocks property must be defined and added to the device tree and the
+      CONFIG_GPIO_HAS_CLOCK_CONTROL configuration option must be enabled.
```
Ok for making `clocks` prop optional but no need to add this Kconfig option.
DTS bindings will ensure clocks is there when required, and DTS macros can be used to make the code depending on the prop presence.

erwango commented on drivers/gpio/Kconfig.stm32:
```diff
@@ -1,11 +1,19 @@
 	help
 	  Enable GPIO driver for STM32 line of MCUs
+
+config GPIO_HAS_CLOCK_CONTROL
+	bool "Enable GPIO clocks handling"
+	default n
+	depends on DT_HAS_ST_STM32MP2_GPIO_ENABLED
+	help
+	  Enable GPIO clocks handling for STM32MP2 series
```
Remore this option. See dts binding comment.

youssefz24 commented:
I just found out that my comments being pending doesn't mean pending for you to review but that they weren't visible at all... ':/ Anyways, I will be waiting for other replies.

youssefz24 commented on soc/st/stm32/stm32mp2x/m33/soc.c:
```diff
@@ -0,0 +1,28 @@
+void soc_early_init_hook(void)
+{
+	/* Update CMSIS SystemCoreClock variable (HCLK) */
+	SystemCoreClock = MHZ(400);
```
If that is the case then wouldn't it be irrelevant for this PR, as the aim is to add the STM32MP2 ?

erwango replied:
Well, that's a bit different for other SoCs in which SystemCoreClock is set to default startup value here and updated to dts value in clock_control driver..
So other SoCs are fine to this regard.

Here it depends on what happen in clock_control register.

erwango replied:
If this depends on external factor (Cortex-A), then indeed it should better be calculated/read in register.

erwango replied:
To conclude on this point, let me get back to the root of what's required:
The essential part on sys clock sw configuration for Zephyr is to correctly configure `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC` (Because CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME is kept disabled).
With current code, this is done today statically through dts `clock-freq` property. If we decide this is the right choice, then nothing more is actually required:
`SystemCoreClock` could be used on other SoC to take into account difference of clock configuration between initial boot and final configuration when exiting clock_control driver (to configure flash latency and so on), but it is not used otherwise.
Since there is actually no configuration done in your case, we probably don't need to update `SystemCoreClock`. So probably the `soc_early_init_hook` could be removed.

youssefz24 replied:
I think I get what you're saying. I removed the `soc_early_init_hook` and soc.c file while at it. 
I do mention in the commit message that the system clock is configuered statically using the dts value. 

Hope that s what you intended ?

arnopo reviewed (commented):
Few new remarks, else main concerns have been addressed, thanks!

nordicjm reviewed (dismissed):
Board files OK

youssefz24 committed:
drivers: interrupt_controller: change dt macro

Replace the use of `DT_NODELABEL(exti)` which depends on the node label
with the defined EXTI_NODE macro using the instance with the
`st_stm32_exti` compatible.
Since both macros point to the same node, this change doesn't affect
the code logic, but makes it independent of the node label, in
preparation for the addition of the STM32MP2 exti nodes.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 commented on drivers/gpio/gpio_stm32.c:
```diff
@@ -349,6 +350,14 @@ static void gpio_stm32_configure_raw(const struct device *dev, gpio_pin_t pin,
+	 * so we do not need to handle them here.
+	 */
+	return 0;
+#else
```
yes you re right. A more clean way in  my opinion is to remove the function `gpio_stm32_init` depending on the presence of the clocks property, and making the function into a `__maybe_unused` to remove the compiler warnings. (see the proposed changes) :D

youssefz24 commented on boards/st/stm32mp257f_ev1/stm32mp257f_ev1_stm32mp257fxx_m33.dts:
```diff
@@ -15,6 +17,36 @@
+			label = "User 1";
+			gpios = <&gpiod 2 GPIO_ACTIVE_LOW>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
```
Done. 
But I just want to point out that the pins referenced in the (User Manual of the STM32MP257F-EV1 board)[https://www.st.com/en/evaluation-tools/stm32mp257f-ev1.html#] uses pin PA13, which differs from the wiki (and the real pin of the button) PG8.  I didn't find an email address to report the error, so perhaps you could point me to one or check the error to avoid future mishaps.

erwango commented on drivers/gpio/gpio_stm32.c:
```diff
@@ -737,12 +738,16 @@ static int gpio_stm32_init(const struct device *dev)
 	PM_DEVICE_DT_DEFINE(__node, gpio_stm32_pm_action);		       \
 	DEVICE_DT_DEFINE(__node,					       \
-			    gpio_stm32_init,				       \
+			    COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),      \
+			    (gpio_stm32_init),				       \
+			    (NULL)),					       \
```
Readability:
```suggestion
			    COND_CODE_1(DT_NODE_HAS_PROP(__node, clocks),      \
					(gpio_stm32_init), (NULL)),	       \
```
(At least indentation is required, use of one line is as you prefer)

youssefz24 replied:
I think it better to leave each of the results of the `COND_COND` in a separate line, in a `if else endif` fashion.

erwango commented on dts/bindings/clock/st,stm32mp2-rcc.yaml:
```diff
@@ -0,0 +1,25 @@
+  enabling/disabling the clocks for the peripherals, when allowed by the
+  Resource Isolation Framework(RIF) peripheral, configured by the cortex-A35
+  core.
+
```
An example on how to declare `clocks` properties would be valuable

erwango reviewed (changes requested):
Few last comments

youssefz24 commented on dts/bindings/gpio/st,stm32mp2-gpio.yaml:
```diff
@@ -0,0 +1,46 @@
+compatible: "st,stm32mp2-gpio"
+
+include:
+  - name: gpio-controller.yaml
```
That was actually what I did first but was advised to redefine the yaml instead.
"you should probably redifine the st, stlm32-gpio.gpio removing the required property for the clcok"
https://github.com/zephyrproject-rtos/zephyr/pull/90295#discussion_r2121474114
So should I stick with the current state or go back to including the `st,stm32-gpio.yaml` and put clocks in property-blocklist as you advised ? 
Kinda lost here :/

erwango replied:
Perfect, txs.

arnopo reviewed (dismissed):
No further comments from my side, it looks ready to go
Thanks for this very nice work @youssefz24 !

youssefz24 commented on dts/arm/st/mp2/stm32mp2_m33.dtsi:
```diff
@@ -56,6 +57,96 @@
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x44240000 DT_SIZE_K(1)>;
+			};
```
I prefer them disabled. less initialization code ;)

erwango commented on west.yml:
```diff
@@ -240,7 +240,7 @@ manifest:
         - hal
     - name: hal_stm32
-      revision: 02d82ee3aa0daa29ab6f107f9969508b89bc6bf7
+      revision: pull/282/head
```
PR has been merged, please update this.
(You'll probably need a rebase before that)

erwango reviewed (changes requested):
Last change and we're good to go!

youssefz24 commented:
@erwango It seems I keep failing the CI twister test 7, when tracing back the issue it seems that it is linked to `hwinfo`.
Also, it builds psa crypto tests even thought I only have GPIO and LED as supported. any ideas how can I disable/skip those unrelated tests ? Or should I try to make the necessary additions ? 
Sorry if my question seems a bit mundane. :/

KozhinovAlexander commented on dts/arm/st/mp2/stm32mp2_m33.dtsi:
```diff
@@ -0,0 +1,161 @@
+			reg = <0x44200000 DT_SIZE_K(64)>;
+		};
+
+		exti2: interrupt-controller@46230000 {
```
@erwango FYI: I see second exti instance. In #85508 I assumed one instance only. Are you agree here EXTI driver implementation shall be updated in #85508?

mathieuchopstm replied:
I have **not** checked the MP2 RefMan at all, but I have been told a few words about this dual-EXTI point, and from my understanding only a single instance is """visible"""[1] in a given context, i.e. the issue is not multi-instance but merely knowing which instance we're operating on. (Ignore my comment if my understanding is wrong)

[1] Not literally as in "not mapped in address space", but in the sense that only one instance is wired to a given CPU

KozhinovAlexander replied:
Thank you both for very fast feedback. Let's stick then with current implementations and see, if extension to multi ecti nodes will be needed later.
From my side I am planning to integrate MP2 dk1 later based on this PR.

borneoa commented on boards/st/stm32mp257f_ev1/support/openocd_stm32mp257f_ev1_m33.cfg:
```diff
@@ -0,0 +1,10 @@
+set EN_CM33   	1
+set EN_CM0P 	0
+
+source [find board/stm32mp25x_dk.cfg]
```
The OpenOCD project will enter in code freeze in ~2 weeks with the target to tag v1.0.0 before the end of the year.
The scripts for STM32MP25 would be inside.
I expect Zephyr to update to the new OpenOCD v1.0.0 once tagged, getting all the new SoC support.
In mean time, having the link to the ST OpenOCD fork looks ok.

youssefz24 committed:
drivers: hwinfo_stm32.c: add support for stm32mp2

Add support for STM32MP2 series in the hwinfo driver.
The STM32MP2 series uses the HAL functions since it doesn't have a
ll_utils.

Signed-off-by: Youssef Zini <youssef.zini@savoirfairelinux.com>

youssefz24 commented:
So to correct the CI twister test, I added the board support in the hwinfo driver. This requires changing the hal file used in the HAL to get the generic header and not the conf one, I opened a pull request in the [STM32_HAL](https://github.com/zephyrproject-rtos/hal_stm32/pull/286) repo for this fix. Since the MP2 doesn t havea utils.h and uses HAL functions.

arnopo commented:
@youssefz24 : seems that you need to rebase on main to resolve conflicts

youssefz24 commented:
Rebased (again) to stay up to date with the main branch :/

youssefz24 commented:
Hello there, 

Thank you all for the reviews and the merge. :tophat:

Youssef Zini
